### PR TITLE
Fixes #1737 and enables to update userHeight

### LINKED ISSUE #1737: camera.userHeight cannot be updated
**Description:**

Currently, changing camera height after camera initialization does not update the offset.

This was made apparent with the `<a-camera>` primitive that first sets the default userHeight to 1.6, and then mappings update the camera after that.

We should fix primitives such that they gather all data before initializing the component, but we could also fix the camera user height updating.

- A-Frame Version:
- Platform/Device:

### PR DESCRIPTION
**Pull request by mkungla**
Added camera.updateHeightOffset so that `camera.userHeight` and attr `user-height` can be updated when headset is not connected without conflicting `removeHeightOffset` and `addHeightOffset`

### CODE REVIEW

File: src/components/camera.js [hunk 1]
```diff
@@ -87,6 +88,30 @@ module.exports.Component = registerComponent('camera', {
...
+  /**
+   * Enables updating camera.userHeight
+   */
+  updateHeightOffset: function () {
+    var currentPosition;
+    var headsetConnected;
+    var el = this.el;
+    var oldHeightOffset = this.userHeightOffset;
+
+    headsetConnected = this.headsetConnected || checkHeadsetConnected();
```
Comment (ngokevin) [inline src/components/camera.js]: Could move this to the block where it is used:

```js

// Do not update camera.userHeight if headset is connected.
headsetConnected = this.headsetConnected || checkHeadsetConnected();
if (headsetConnected) { return; }

```

File: src/components/camera.js [hunk 2]
```diff
@@ -87,6 +88,30 @@ module.exports.Component = registerComponent('camera', {
...
   },
 
+  /**
+   * Enables updating camera.userHeight
+   */
+  updateHeightOffset: function () {
+    var currentPosition;
+    var headsetConnected;
+    var el = this.el;
+    var oldHeightOffset = this.userHeightOffset;
```
Comment (ngokevin) [inline src/components/camera.js]: Should we get rid of `this.userHeightOffset`? This is always available at `this.data.userHeight` and the old height is available through `update (oldData)`, and we don't want extra state.

**Comment by ngokevin** [comment]
Could you add a test in `tests/components/camera.test.js` to test the offset is updated? Thanks!

File: src/components/camera.js (hunk 2 above)
Comment (mkungla) [inline src/components/camera.js]: agree `userHeightOffset` is superfluous

**Comment by ngokevin** [comment]
Need a proper rebase.

**Comment by mkungla** [comment]
yes I see

**Comment by ngokevin** [comment]
What steps were you trying to rebase? Might be good to add to CONTRIBUTING.md guide.

Should be `git pull --rebase origin master` (if origin points to this aframevr/aframe upstream).

**Comment by mkungla** [comment]
I had to rewind it that why that got closed.

**Comment by ngokevin** [comment]
Hm, I usually never have to close a PR. Just want to know so I can document. I usually do `git rebase -i HEAD~10` to squash.

**Comment by mkungla** [comment]
just do `git rebase upstream/,master` !
this one here got closed since rewinded and push state which was alredy merged github auto closes and marks it merged.

**Comment by mkungla** [comment]
in my case I have `git remote -v`
```bash
origin	git@github.com:mkungla/aframe.git (fetch)
origin	git@github.com:mkungla/aframe.git (push)
upstream	git@github.com:aframevr/aframe.git (fetch)
upstream	git@github.com:aframevr/aframe.git (push)
```

so what ever branch you want to rebase just run
`git fetch --all` and
`git rebase upstream/master`

**Pull request closed without merging**